src/hotspot/cpu/aarch64/nativeInst_aarch64.cpp

```diff
@@ -104,22 +103,19 @@ void NativeCall::set_destination_mt_safe(address dest, bool assert_lock) {
 }
 
 address NativeCall::get_trampoline() {
-  address call_addr = addr_at(0);
+  address call_addr = instruction_address();
 
   CodeBlob *code = CodeCache::find_blob(call_addr);
   assert(code != nullptr, "Could not find the containing code blob");
+  if (!code->is_nmethod()) return nullptr;
```

Comment:
Is it because trampolines are not used by not nmethods? Should we have an assert here about that?

Reply:
Previously, get_trampoline() was called on CodeBuffer/CodeBlob during expansion, but now it is only called by set_destination_mt_safe, so it should be OK to assert that it is an nmethod.

Reply:
Good.

Reply:
> trampoline

See docs on different types of calls AArch64 uses: https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp#L1183

Reply:
Trampoline calls are only used in nmethods because they need a stub code section.